### PR TITLE
Use clang format for code fomratting

### PR DESCRIPTION
**Pull request by ktaborowski**
.clang-format configuration file from sdk-nrf repository

### CODE REVIEW

File: .github/workflows/on_PR.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
 
       - name: Install requirements
         run: |
-          python3 -m pip install -r ./scripts/ci/requirements.txt && sudo apt install uncrustify
+          python3 -m pip install -r ./scripts/ci/requirements.txt && sudo apt-get install clang-format
```
Comment (RobertGalatNordic) [inline .github/workflows/on_PR.yml:21]: From my experience clang-format behaves differently on different version, we should consider settling for one specific version of clang-format

File: utils/src/state_notifier_gpio_backend.c [hunk 2]
```diff
@@ -21,11 +21,12 @@ static inline struct gpio_pin state_to_pin_mapper(enum application_state state)
 {
 	/* *INDENT-OFF* */
```
Comment (RobertGalatNordic) [inline utils/src/state_notifier_gpio_backend.c:22]: This comment is only relevant for uncrustify, it should be removed or replaced with clang-format comment to disable formatting

File: subsys/sal/sid_pal/src/sid_mfg_storage.c [hunk 3]
```diff
@@ -60,48 +60,60 @@ struct sid_pal_mfg_store_value_to_address_offset {
 static void ntoh_buff(uint8_t *buffer, size_t buff_len);
 static uint32_t default_app_value_to_offset(int value);
 static off_t checked_addr_return(off_t offset, uintptr_t start_address, uintptr_t end_address);
-static off_t value_to_offset(sid_pal_mfg_store_value_t value, uintptr_t start_address, uintptr_t end_address);
+static off_t value_to_offset(sid_pal_mfg_store_value_t value, uintptr_t start_address,
+			     uintptr_t end_address);
 
 // *INDENT-OFF*
```
Comment (RobertGalatNordic) [inline subsys/sal/sid_pal/src/sid_mfg_storage.c:66]: This comment is only relevant for uncrustify, it should be removed or replaced with clang-format comment to disable formatting

File: utils/src/state_notifier_gpio_backend.c (hunk 2 above)
Comment (ktaborowski) [reply utils/src/state_notifier_gpio_backend.c]: Done

File: subsys/sal/sid_pal/src/sid_mfg_storage.c (hunk 3 above)
Comment (ktaborowski) [reply subsys/sal/sid_pal/src/sid_mfg_storage.c]: Done

File: samples/sensor_monitoring/include/sm_utils.h [hunk 4]
```diff
@@ -11,25 +11,27 @@
...
 #error "Not defined Sidewalk link mask!!"
 #endif
 
-#define SID_APP_LM_2_STR(_lm)   (_lm == SID_LINK_TYPE_1 ? "BLE" :			      \
-				 (_lm == SID_LINK_TYPE_2) ? "FSK" :			      \
-				 (_lm == SID_LINK_TYPE_3) ? "LoRa" :			      \
-				 (_lm == (SID_LINK_TYPE_1 | SID_LINK_TYPE_2)) ? "BLE & FSK" : \
-				 (_lm == (SID_LINK_TYPE_1 | SID_LINK_TYPE_3)) ? "BLE & LoRa" : "INVALID")
+#define SID_APP_LM_2_STR(_lm)                                                                      \
+	(_lm == SID_LINK_TYPE_1			      ? "BLE" :                                    \
```
Comment (RobertGalatNordic) [inline samples/sensor_monitoring/include/sm_utils.h:29]: this looks like 1 space less than in other lines

**Review by RobertGalatNordic — APPROVED** [review]

**Pull request merged**